rollup.config.js

```diff
@@ -24,10 +24,15 @@ const ts_plugin = is_publish
 	  });
 
 fs.writeFileSync(
-	`./compiler.d.ts`,
+	'./compiler.d.ts',
 	`export { compile, parse, preprocess, walk, VERSION } from './types/compiler/index.js';`
 );
 
+fs.writeFileSync(
+	'./src/shared/version.js',
+	`/** @type {string} */\nexport const VERSION = '${pkg.version}';`
+);
+
```

Comment:
Since the build is run as part of the `prepare` script, this will be missing only those who already have the repository checked out - doing `pnpm build` once is fine for me in that case. It's the same for the `internal_exports.js` btw which exists for a long time already. I wouldn't want those things to be checked in personally.

Reply:
`preblishOnly` triggers the build which contains the `version.js` generation as part of it, at which point the pkg.version has already been updated, so it's safe to do it that way.

Reply:
personally I don't think the stuff the version is used for is that important and I'd be fine to just drop it as it doesn't really seem necessary and people can read or import the `package.json` to get the version themselves if they really care about it